src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1362,9 +1342,8 @@ public override BoundNode VisitArrayAccess(BoundArrayAccess node)
             VisitRvalue(node.Expression);
 
             Debug.Assert(!IsConditionalState);
-            // No need to check expression type since System.Array is a reference type.
             Debug.Assert(!node.Expression.Type.IsValueType);
-            CheckPossibleNullReceiver(node.Expression, checkType: false);
```

Comment:
>CheckPossibleNullReceiver [](start = 12, length = 25)

Not related to this PR: I looked at `CheckPossibleNullReceiver` to see what `checkType` means and it's not obvious. Maybe add an xml doc there or rename if you have a better idea?